kedro-org/kedro-viz
# Conditionally move session store and stats file to .viz directory

## Description

Resolves #1891 

## Development notes

- Created 2 constants at `package/kedro_viz/constants.py` - 

```python
VIZ_SESSION_STORE_ARGS = {"path": ".viz"}
```

```python
VIZ_METADATA_ARGS = {"path": ".viz"}
```

- If the user provides SESSION_STORE_ARGS, Kedro-Viz will use it to store the session_store.db file. If not, it will use the VIZ_SESSION_STORE_ARGS["path"] .
- Started writing/reading stats.json to/from VIZ_METADATA_ARGS["path"] with the new release.
- Updated document about the default VIZ_SESSION_STORE_ARGS
- Updated document about `stats.json` being created under `.viz` if they have viz installed and did not disable hooks for `kedro-viz` plugin.

## QA notes

- From your kedro project, run `kedro run` by defining  `SESSION_STORE_ARGS` in your settings.py file. `session_store.db` file will be modified/created at the location specified in settings.py file.
- `kedro viz run` should read sessions data for experiment tracking from the location specified above
- From your kedro project, run `kedro run` without defining  `SESSION_STORE_ARGS` in your settings.py file. `session_store.db` file will be modified/created at `<kedro_project_root>/.viz` directory.
- `kedro viz run` should read sessions data for experiment tracking from `<kedro_project_root>/.viz/session_store.db`
- With `kedro run` `stats.json` file will be created under `<kedro_project_root>/.viz` directory
- `kedro viz run` should read data of stats.json from the new directory created above

**NOTE:**
1. Kedro starters will not be modified in this PR.
2. Users need to copy their `session_store.db` file manually to .viz folder in-case they want to remove `SESSION_STORE_ARGS` from their settings.py file and would like to have the `session_store.db` file created under `.viz` folder for future kedro runs. They can ignore this if they do not wish to either use `.viz` folder or do not want historic `kedro run` details in experiment tracking.

## Checklist
